src/server/kernel/wseat.cpp

```diff
@@ -71,6 +73,25 @@ class WSeatWheelEvent : public QWheelEvent {
 };
 #endif
 
+class WSeatHoldEvent : public QInputEvent {
+public:
+    static const QEvent::Type HoldBegin = static_cast<QEvent::Type>(2000);
+    static const QEvent::Type HoldEnd = static_cast<QEvent::Type>(2001);
+    explicit WSeatHoldEvent(QEvent::Type type, const QPointingDevice *dev, int fingers = 1)
+        : QInputEvent(HoldBegin, dev)
+        , m_fingers(fingers)
+    {
+    }
+
+    inline int fingerCount() const { return m_fingers; }
+    inline void enableCancelled() { m_cancelled = true; }
+    inline bool cancelled() const { return m_cancelled; }
+
+private:
+    int m_fingers = 0;
```

Comment:
m_fingers default value should same in constructor.
`explicit WSeatHoldEvent(QEvent::Type type, const QPointingDevice *dev, int fingers = 1)`